<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>551</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>550</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>551</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Генератор Лабиринтов и Пещер</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_7">
    <item>
     <widget class="QGraphicsView" name="mainGraphicView">
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
      <property name="sceneRect">
       <rectf>
        <x>0.000000000000000</x>
        <y>0.000000000000000</y>
        <width>500.000000000000000</width>
        <height>500.000000000000000</height>
       </rectf>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QRadioButton" name="mazeRadioButton">
          <property name="text">
           <string>Лабиринт</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="cavesRadioButton">
          <property name="text">
           <string>Пещера</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="pageMaze">
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_2">
                <item>
                 <widget class="QLabel" name="labelMazeWidth">
                  <property name="text">
                   <string>Ширина:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="spinBoxMazeWidth">
                  <property name="minimumSize">
                   <size>
                    <width>110</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>111</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="value">
                   <number>10</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout">
                <item>
                 <widget class="QLabel" name="labelMazeHeight">
                  <property name="text">
                   <string>Высота:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="spinBoxMazeHeight">
                  <property name="minimumSize">
                   <size>
                    <width>110</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>111</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="value">
                   <number>10</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="generateMazeButton">
              <property name="text">
               <string>Сгенерировать</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QPushButton" name="loadMazeButton">
                <property name="text">
                 <string>Загрузить...</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="saveMazeButton">
                <property name="text">
                 <string>Сохранить...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>300</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageCave">
         <layout class="QVBoxLayout" name="verticalLayout_14">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_4">
                <item>
                 <widget class="QLabel" name="labelCavesWidth">
                  <property name="text">
                   <string>Ширина:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="spinBoxCavesWidth">
                  <property name="minimumSize">
                   <size>
                    <width>110</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>111</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="value">
                   <number>20</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_5">
                <item>
                 <widget class="QLabel" name="labelCavesHeight">
                  <property name="text">
                   <string>Высота:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="spinBoxCavesHeight">
                  <property name="minimumSize">
                   <size>
                    <width>110</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>111</width>
                    <height>21</height>
                   </size>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="value">
                   <number>20</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="generateCavesButton">
              <property name="text">
               <string>Сгенерировать</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="loadCavesButton">
                <property name="text">
                 <string>Загрузить...</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="saveCavesButton">
                <property name="text">
                 <string>Сохранить...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QLabel" name="labelBirthLimit">
                <property name="text">
                 <string>Предел рождения</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBoxBirthLimit">
                <property name="minimumSize">
                 <size>
                  <width>110</width>
                  <height>21</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>111</width>
                  <height>21</height>
                 </size>
                </property>
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>7</number>
                </property>
                <property name="value">
                 <number>4</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QLabel" name="labelDeathLimit">
                <property name="text">
                 <string>Предел смерти</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBoxDeathLimit">
                <property name="minimumSize">
                 <size>
                  <width>110</width>
                  <height>21</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>111</width>
                  <height>21</height>
                 </size>
                </property>
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="maximum">
                 <number>7</number>
                </property>
                <property name="value">
                 <number>4</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Вероятность рождения живых клеток</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QSlider" name="live_dead_cells_rate_slider">
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>100</number>
                </property>
                <property name="value">
                 <number>50</number>
                </property>
                <property name="tracking">
                 <bool>true</bool>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::TicksAbove</enum>
                </property>
                <property name="tickInterval">
                 <number>25</number>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <widget class="QLabel" name="label_8">
                  <property name="text">
                   <string>0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_9">
                  <property name="text">
                   <string>100</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_13">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QPushButton" name="oneStepButton">
                <property name="text">
                 <string>Один шаг</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="autoStepButton">
                <property name="text">
                 <string>Авто</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QSpinBox" name="spinBoxDelay">
                <property name="minimum">
                 <number>100</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="singleStep">
                 <number>50</number>
                </property>
                <property name="value">
                 <number>250</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>миллисекунд</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>72</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
